lib/board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+
+  attr_accessor :cells #has an attribute to store the cells of the board
+
+  def initialize #sets the cells of the board to a 9 element array of " "
+    @cells = Array.new(9," ")
+  end
+
+  def reset! #can reset the state of the cells in the board; sets the cells of the board to a 9 element array of " "
+    self.cells.clear
+    @cells = Array.new(9," ")
+  end
+
+  def display
+   puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+   puts "-----------"
+   puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+   puts "-----------"
+   puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def index(position) #converts the user's inputted position into its corresponding index within @cells
+    position.to_i - 1
+  end
+
+  def position(position) #takes in user input and returns the value of the board cell'
+    @cells[index(position)]
+  end
+
+  def full? #returns true for a full board; returns false for an in-progress game
+    @cells.each do |cell|
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.